runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/PipelineValidator.java

```diff
@@ -221,23 +229,39 @@ private static void validateParDo(String id, PTransform transform, Components co
           id,
           sideInputId);
     }
-    // TODO: Validate state_specs and timer_specs
+    if (payload.getStateSpecsCount() > 0 || payload.getTimerSpecsCount() > 0) {
```

Comment:
This should also have timer family specs.

@boyuanzz Once this code is merged, this is another place where TimerSpecs will need to be removed.

Reply:
Ack. can we merge it as soon as possible? Then I can rebase my changes on top of this.

Reply:
We can merge now and you will need to swap from TimerSpecs to TimerFamilySpecs here as part of your PR.